pilot/pkg/networking/apigen/apigen.go

```diff
@@ -126,7 +126,7 @@ func (g *APIGenerator) Generate(proxy *model.Proxy, w *model.WatchedResource, re
 			if s.Attributes.ServiceRegistry == provider.External {
 				continue
 			}
-			c := serviceentry.ServiceToServiceEntry(s, proxy)
+			c := serviceentry.ServiceToServiceEntry(s, proxy, req.Push)
```

Comment:
Taking a step back... what is the point of this? What use is a Service without the endpoints?

Reply:
Let me explain my use case so we can understand why we need target port in the service entry:

To support non-HTTP protocols in Istio, Aeraki retrieves service entries from Istio via xds, using service entries to generate envoyfilters to replace the original TCP filter in the outbound and inbound listeners. Here is an example of an envoyfilter for inbound listeners:

```
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  labels:
    manager: Aeraki
  name: aeraki-inbound-thrift-sample-server.meta-thrift.svc.cluster.local-9090
  namespace: istio-system
spec:
  configPatches:
  - applyTo: NETWORK_FILTER
    match:
      listener:
        filterChain:
          destinationPort: 9090
          filter:
            name: envoy.filters.network.tcp_proxy
        name: virtualInbound
    patch:
      operation: REPLACE
      value:
        name: envoy.filters.network.meta_protocol_proxy
        typed_config:
          '@type': type.googleapis.com/udpa.type.v1.TypedStruct
          type_url: type.googleapis.com/aeraki.meta_protocol_proxy.v1alpha.MetaProtocolProxy
          value:
            applicationProtocol: thrift
            codec:
              name: aeraki.meta_protocol.codec.thrift
            metaProtocolFilters:
            - config:
                '@type': type.googleapis.com/aeraki.meta_protocol_proxy.filters.local_ratelimit.v1alpha.LocalRateLimit
                conditions:
                - match:
                    metadata:
                    - exactMatch: sayHello
                      name: method
                  tokenBucket:
                    fillInterval: 10s
                    maxTokens: 2
                    tokensPerFill: 2
                statPrefix: thrift-sample-server.meta-thrift.svc.cluster.local
                tokenBucket:
                  fillInterval: 60s
                  maxTokens: 5
                  tokensPerFill: 5
              name: aeraki.meta_protocol.filters.local_ratelimit
            - name: aeraki.meta_protocol.filters.router
            routeConfig:
              name: inbound|9090||
              routes:
              - route:
                  cluster: inbound|9090||
            statPrefix: inbound|9090||
  workloadSelector:
    labels:
      app: thrift-sample-server
```

Since we don't have target ports in the service entry, Aeraki is using the service port to generate the inbound envoyfilter(the destinationPort match condition in the envoyfilter), which is just a workaround and won't replace the original TCP filter in case the target port number is not the same as the service port.

The reported issue in Aeraki: https://github.com/aeraki-mesh/aeraki/issues/194

Reply:
> Taking a step back... what is the point of this? What use is a Service without the endpoints?

@howardjohn I added the req.Push since commit: `cf4a022`, so should I step back to `ae8fae` ?